src/main/java/com/fleotadezuta/youthprogrammanager/facade/ChildParentFacade.java

```diff
@@ -0,0 +1,211 @@
+                        .collectList()
+                        .map(children -> {
+                            var parentWithChildrenDto = parentMapper.fromParentDocumentToParentWithChildrenDto(parent);
+                            List<ChildDto> childDtos = children.stream()
+                                    .map(childMapper::fromChildDocumentToChildDto)
+                                    .collect(Collectors.toList());
+                            parentWithChildrenDto.setChildDtos(childDtos);
+                            return parentWithChildrenDto;
+                        }));
+    }
+
+    public Mono<ParentDto> addParent(ParentCreateDto parentCreateDto) {
+        var parentDto = parentMapper.fromParentCreateDtoToParentDto(parentCreateDto);
+        return parentService.validateParent(parentDto)
+                .map(parentMapper::fromParentDtoToParentDocument)
+                .flatMap(parentService::save)
+                .flatMap(validatedParent -> {
+                    String childId = parentCreateDto.getChildId();
+                    if (childId == null || childId.isEmpty()) {
+                        return Mono.just(validatedParent);
+                    } else {
+                        return childService.findById(childId)
+                                .flatMap(childDocument -> {
+                                    var relativeParents = childDocument.getRelativeParents();
+                                    relativeParents.add(new RelativeParents(validatedParent.getId(), true));
+                                    childDocument.setRelativeParents(relativeParents);
+                                    return childService.save(childDocument)
+                                            .thenReturn(validatedParent);
+                                });
+                    }
+                })
+                .map(parentMapper::fromParentDocumentToParentDto);
+    }
+
+
+    public Mono<ParentDto> updateParent(ParentUpdateDto parentUpdateDto) {
+        return getParentById(parentUpdateDto.getId())
+                .flatMap(previousParent -> {
+                    var parentDto = parentMapper.fromParentUpdateDtoToParentDto(parentUpdateDto);
+                    return Mono.just(parentDto)
+                            .flatMap(parentService::validateParent)
+                            .map(parentMapper::fromParentDtoToParentDocument)
+                            .flatMap(parentDoc -> {
+                                parentDoc.setId(parentDto.getId());
+                                List<String> childIds = parentUpdateDto.getChildIds();
+                                List<String> previousChildIds = previousParent.getChildDtos().stream()
+                                        .map(ChildDto::getId)
+                                        .toList();
+                                Set<String> combinedChildIds = new HashSet<>(childIds);
+                                combinedChildIds.addAll(previousChildIds);
```

Comment:
Ha a korábbi gyerek ID-kat mindig visszatesszük a Set-be, hogyan lehet törölni egy gyereket a parent oldaláról?

Reply:
!childIds.contains(child.getId()) && previousChildIds.contains(child.getId()
ha az updateben lévő List be nincs benne de eddig hozzá tartozott a gyerek akkor töröljük